sw/device/silicon_creator/lib/ownership/owner_block.c

```diff
@@ -229,19 +235,47 @@ rom_error_t owner_block_flash_apply(const owner_flash_config_t *flash,
           .erase = bitfield_field32_read(val, FLASH_CONFIG_ERASE),
       };
 
-      if (lockdown == config_side) {
-        if (bitfield_field32_read(val, FLASH_CONFIG_PROTECT_WHEN_PRIMARY) !=
-            kMultiBitBool4False) {
+      uint32_t pwp =
+          bitfield_field32_read(val, FLASH_CONFIG_PROTECT_WHEN_PRIMARY);
+      hardened_bool_t lock =
+          bitfield_field32_read(val, FLASH_CONFIG_LOCK) != kMultiBitBool4False
+              ? kHardenedBoolTrue
+              : kHardenedBoolFalse;
+      uint32_t region_start = config->start;
+      uint32_t region_end = region_start + config->size;
+
+      // When applying the flash configuration, a region is a ROM_EXT region if
```

Comment:
Thank you for the detailed comment! The comment and the logic don't quite match either other. I agree with the comment. The logic to match the comment would be

```
kRomExtAStart <= region_start < kRomExtAEnd &&
kRomExtAStart < region_end <= kRomExtAEnd
```

(and maybe we need an explicit `region_start<=region_end` check too?)

Reply:
:facepalm: How embarrassing.  A total logic fail.

I've fixed this and added unit tests to `owner_block_unittest.cc` to test the overlap/exclusivity checking functions.

We don't need to test `region_start <= region_end` because the bounds are computed from a start and size (which are `uint16_t`) and the `region_*` values are `uint32_t`.

I added a sanity check to the `end` value: if the end is beyond the total size of the flash, the configuration is invalid.